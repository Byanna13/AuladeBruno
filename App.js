/*Importa os codigos das outras paginas*/
import { View } from "react-native";
import Stack from "./src/navigation/Stack";
import { NavigationContainer } from "@react-navigation/native";

export default () => {
  return (
    <View>
      <NavigationContainer>
        <Stack />
      </NavigationContainer>
    </View>
  )
}