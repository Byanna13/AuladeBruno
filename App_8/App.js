import { Text, View } from "react-native";
import { NavigationContainer } from "@react-navigation/native";
import { createNativeStackNavigator } from "@react-navigation/native-stack";
import Index from './src/pages/Index';
import Contato from './src/pages/Contato';
import Sobre from './src/pages/Sobre';

const Stack = createNativeStackNavigator()

export default () => {
  return (
    <NavigationContainer>
      {/*screenOptions={{headerShown: false}} retira o cabeçalho, e igual a true coloca de volta, sem o codigo ele fica automaticamente*/}
      <Stack.Navigator initialRouteName="Index" screenOptions={{headerShown: true}} >
        {/*options={{ title: 'TELA INICIAL' define o nome do cabeçalho*/}
        {/*Add estilo usando headerStyle:{conteudos}*/}
      <Stack.Screen name="Index" component={Index} options={{ title: 'TELA INICIAL',
            headerStyle: {
            backgroundColor: 'green', }, headerTintColor: '#fff',
          headerTitleStyle: {
            fontWeight: 'bold',
          },
            }} />
      <Stack.Screen name="Contato" component={Contato} options={{ title: 'CONTATO',
            headerStyle: {
            backgroundColor: 'green', }, headerTintColor: '#fff',
          headerTitleStyle: {
            fontWeight: 'bold',
          }, }} />
      <Stack.Screen name="Sobre" component={Sobre} options={{ title: 'SOBRE',
            headerStyle: {
            backgroundColor: 'green', }, headerTintColor: '#fff',
          headerTitleStyle: {
            fontWeight: 'bold',
          }, }} />
      </Stack.Navigator>
    </NavigationContainer>
  )
}