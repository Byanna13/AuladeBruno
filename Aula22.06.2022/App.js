import { Text, View, TextInput, Button } from 'react-native';

export default () => {
  return (
    <View>
      <Text testID='titulo'>Digite um n√∫mero</Text>
      <TextInput testID='campo' placeholder="Digite aqui" />
      <Button testID='botao' title="Calcular" />
    </View>
  );
}
/*Para testar o codigo usa [npm run test -u], para testar e atualizar o codigo usa [npm run test -- -u] no prompt
[testID] para identificar os componentes pelo 'id'
[testText] para indentificar os componentes pelo 'texto'
*/