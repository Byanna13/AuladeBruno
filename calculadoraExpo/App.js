import * as React from 'react';
import { Text, View, StyleSheet } from 'react-native';
import Constants from 'expo-constants';

// You can import from local files
import Botoes from './components/Botoes';

// or any pure javascript modules available in npm
import { Card } from 'react-native-paper';

export default function App() {
  return (
    <View style={styles.container}>
      <Text style={styles.paragraph}>
        Calculadora
      </Text>
      <Card>
        <Botoes style={styles.but} />
      </Card>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    paddingTop: Constants.statusBarHeight,
    backgroundColor: 'yellow',
    padding: 8,
  },
  paragraph: {
    margin: 24,
    fontSize: 18,
    fontWeight: 'bold',
    textAlign: 'center',
  },
});
