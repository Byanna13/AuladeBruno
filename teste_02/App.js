import { Text, View, TextInput, Button } from 'react-native';
import { useState } from 'react';

export default () => {
  const [numero, setNumero] = useState(0);
  const [dobro, setDobro] = useState(0);

  function dobro2(){
    setDobro(parseInt(numero)*2)
  }

  return (
    <View>
      <Text>Cálculo do Dobro</Text>
      <TextInput testID="entrada" placeholder='Digite um número: ' onChangeText={setNumero}/>
      <Button title="Calcular" onPress={dobro2} testID="calc" />
      <Text testID="resultado">{dobro}</Text>
    </View>
  );
}
