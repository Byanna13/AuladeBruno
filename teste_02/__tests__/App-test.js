import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react-native'
import App from '../App';

describe('Teste de Snapshot', () => {
    it('Snapshot de App', () => {
        const tela = renderer.create(<App />).toJSON();
        expect(tela).toMatchSnapshot();
    })
})
describe('Teste de Componentes', () => {
    it('Snapshot de Componentes de App', () => {
        const {getByText, getByTestId } = render (<App />);

        const textoInicial = getByText('Cálculo do Dobro');
        expect(textoInicial).toBeDefined();
        //expect(textoInicial).toBeTruthy();
        
        expect(getByTestId("entrada")).toBeDefined();
        expect(getByTestId("calc")).toBeDefined();
        expect(getByTestId("resultado")).toBeDefined();
        
    })
})

describe('Testando Operações', () => {
    it('Teste de Dobro Positivo', () => {
        const {getByTestId, getByText, getAllByText} = render (<App />);

        const campoEntrada = getByTestId("entrada");
        fireEvent.changeText(campoEntrada, 15);

        const botao = getByTestId("calc");
        fireEvent.press(botao);

        //expect(getByText("30")).toBeTruthy();
        expect(getAllByText("30").length).toBe(1);
    })
    it('Teste de Dobro Negativo', () => {
        const {getByTestId, getByText, getAllByText} = render (<App />);

        const campoEntrada = getByTestId("entrada");
        fireEvent.changeText(campoEntrada, -15);

        const botao = getByTestId("calc");
        fireEvent.press(botao);

        //expect(getByText("30")).toBeTruthy();
        expect(getAllByText("-30").length).toBe(1);
    })
})