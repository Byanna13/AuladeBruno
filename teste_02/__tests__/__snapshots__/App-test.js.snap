// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teste de Snapshot Snapshot de App 1`] = `
<View>
  <Text>
    Cálculo do Dobro
  </Text>
  <TextInput
    onChangeText={[Function]}
    placeholder="Digite um número: "
    testID="entrada"
  />
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="calc"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Calcular
      </Text>
    </View>
  </View>
  <Text
    testID="resultado"
  >
    0
  </Text>
</View>
`;
